packages/gestalt/src/ButtonToggle.tsx

```diff
@@ -0,0 +1,243 @@
+ *
+ * ![ButtonToggle light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/ButtonToggle.spec.mjs-snapshots/ButtonToggle-chromium-darwin.png)
+ * ![ButtonToggle dark mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/ButtonToggle-dark.spec.mjs-snapshots/ButtonToggle-dark-chromium-darwin.png)
+ *
+ */
+const ButtonWithForwardRef = forwardRef<HTMLButtonElement, Props>(function ButtonToggle(
+  {
+    accessibilityLabel,
+    color = 'transparent',
+    dataTestId,
+    disabled = false,
+    iconStart,
+    onClick,
+    selected = false,
+    size = 'md',
+    text,
+    accessibilityControls,
+  }: Props,
+  ref,
+) {
+  const innerRef = useRef<null | HTMLButtonElement>(null);
+
+  // When using both forwardRef and innerRef, React.useimperativehandle() allows a parent component
+  // that renders <ButtonToggle ref={inputRef} /> to call inputRef.current.focus()
+  // @ts-expect-error - TS2322 - Type 'HTMLButtonElement | null' is not assignable to type 'HTMLButtonElement'.
+  useImperativeHandle(ref, () => innerRef.current);
+
+  const {
+    compressStyle,
+    isTapping,
+    handleBlur,
+    handleMouseDown,
+    handleMouseUp,
+    handleTouchStart,
+    handleTouchMove,
+    handleTouchCancel,
+    handleTouchEnd,
+  } = useTapFeedback({
+    height: innerRef?.current?.clientHeight,
+    width: innerRef?.current?.clientWidth,
+  });
+
+  const { colorSchemeName } = useColorScheme();
+  // We need to make a few exceptions for accessibility reasons in darkMode for red buttons
+  const isDarkMode = colorSchemeName === 'darkMode';
+  const isDarkModeRed = isDarkMode && color === 'red';
+  const { isFocusVisible } = useFocusVisible();
+
+  const sharedTypeClasses = classnames(styles.button, focusStyles.hideOutline, {
+    [focusStyles.accessibilityOutline]: !disabled && isFocusVisible,
+  });
+
+  const baseTypeClasses = classnames(sharedTypeClasses, touchableStyles.tapTransition, {
+    [styles.sm]: size === 'sm',
+    [styles.md]: size === 'md',
+    [styles.lg]: size === 'lg',
+    [styles[color]]: !disabled && !selected,
+    [styles.noBorder]: color === 'red' && !selected && !disabled,
+    [styles.selected]: !disabled && selected,
+    [styles.disabled]: disabled && (color !== 'red' || selected),
+    [styles.disabledRed]: disabled && color === 'red' && !selected,
+    [styles.enabled]: !disabled,
+    [touchableStyles.tapCompress]: !disabled && isTapping,
+  });
+
+  const parentButtonClasses = classnames(sharedTypeClasses, styles.parentButton);
+
+  const childrenDivClasses = classnames(baseTypeClasses, styles.childrenDiv);
+
+  const textColor =
+    (disabled && 'subtle') ||
+    (isDarkModeRed && 'default') ||
+    (selected && 'default') ||
```

Comment:
is the order correct? 

should it be disabled/selected/isDarkRedMode?

Reply:
Fixed, it had ended up like this after some iteration.